safe-svg.php

```diff
@@ -560,19 +560,43 @@ public function metadata_error_fix( $data, $post_id ) {
 		/**
 		 * Get SVG size from the width/height or viewport.
 		 *
-		 * @param string|false $svg The file path to where the SVG file should be, false otherwise.
+		 * @param integer $attachment_id The attachment ID of the SVG being processed.
 		 *
 		 * @return array|bool
 		 */
-		protected function svg_dimensions( $svg ) {
+		protected function svg_dimensions( $attachment_id ) {
+			/**
+			 * Calculate SVG dimensions and orientation.
+			 *
+			 * This filter allows you to implement your own sizing. By returning a non-false
+			 * value, it will short-circuit this function and return your set value.
+			 *
+			 * @param boolean Default value of the filter.
+			 * @param integer $attachment_id The attachment ID of the SVG being processed.
+			 *
+			 * @return array|false An array of SVG dimensions and orientation or false.
+			 */
+			$short_circuit = apply_filters( 'safe_svg_pre_dimensions', false, $attachment_id );
+
+			if ( false !== $short_circuit ) {
+				return $short_circuit;
+			}
+
 			if ( ! function_exists( 'simplexml_load_file' ) ) {
 				return false;
 			}
 
-			$svg    = @simplexml_load_file( $svg ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
-			$width  = 0;
-			$height = 0;
-			if ( $svg ) {
+			$svg      = get_attached_file( $attachment_id );
+			$metadata = wp_get_attachment_metadata( $attachment_id );
+			$width    = 0;
+			$height   = 0;
+
+			if ( $svg && ! empty( $metadata['width'] ) && empty( $metadata['height'] ) ) {
```

Comment:
@sksaju Sorry, I know this is late and this PR is already merged but in looking at this prior to prepping the next release, on the surface it seems like this conditional is not correct. It checks `$svg` isn't `null` and then checks if the width isn't empty and the height is empty. Unless I'm thinking of this wrong, shouldn't it be checking if both width and height aren't empty, so like this:

```php
if ( $svg && ! empty( $metadata['width'] ) && ! empty( $metadata['height'] ) ) {
```

Reply:
@dkotter Yeah it should be `! empty`, I've created a PR https://github.com/10up/safe-svg/pull/174 to fix it, Apologies & Thanks